internal/providers/packet/packet.go

```diff
@@ -48,3 +62,102 @@ func FetchConfig(f resource.Fetcher) (types.Config, report.Report, error) {
 
 	return util.ParseConfig(f.Logger, data)
 }
+
+// PostStatus posts a message that will show on the Packet Instance Timeline
+func PostStatus(f resource.Fetcher, errMsg error) error {
+
+	f.Logger.Info("POST message to Packet Timeline")
+	// fetch JSON from https://metadata.packet.net/metadata
+	data, err := f.FetchToBuffer(metadataUrl, resource.FetchOptions{
+		Headers: nil,
+	})
+	if data == nil {
```

Comment:
Is there a possible case where a `nil` is returned for both `data` & `err`? If so then this function will appear to pass despite not working. Also this check should probably be combined with the `err != nil` check.

Reply:
| Data | Err | return value|
| --- | --- | --- |
| `!nil`  | `!nil` | `err` |
| `!nil` | `nil` | do nothing |
| `nil` | `!nil` | `err` |
| `nil` | `nil` | `ErrValidFetchEmptyData` |

for this:
```
	if err != nil {
		return err
	}
	if data == nil {
		return errors.ErrValidFetchEmptyData
	}
```
internal/providers/packet/packet.go ( line 72 )